### PR TITLE
prevent NoClassDefFound

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
ref: https://github.com/nextcloud/android/issues/541

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
Tested with android 4.1

File: src/com/owncloud/android/datamodel/UploadsStorageManager.java [hunk 1]
```diff
@@ -381,7 +382,11 @@ private OCUpload createOCUploadFromCursor(Cursor c) {
         );
 
         // add pending Jobs
-        return getPendingJobs().toArray(uploads);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            return uploads;
+        } else {
+            return getPendingJobs().toArray(uploads);
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/datamodel/UploadsStorageManager.java]: haven't used it for a while, will have to see if I can test this later, but wouldn't the list elements then override the array items already in there? 
What I would have thought:
```
List<OCUpload> result= getPendingJobs();
Collections.addAll(result, uploads);
result.toArray(result.size())
```

File: src/com/owncloud/android/datamodel/UploadsStorageManager.java [hunk 2]
```diff
@@ -381,7 +382,11 @@ private OCUpload createOCUploadFromCursor(Cursor c) {
         );
 
         // add pending Jobs
-        return getPendingJobs().toArray(uploads);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/datamodel/UploadsStorageManager.java]: JobScheduler is API level 21 (Lollipop): https://developer.android.com/reference/android/app/job/JobScheduler.html

File: src/com/owncloud/android/datamodel/UploadsStorageManager.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/datamodel/UploadsStorageManager.java]: Right. I will change it.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky review done, tested successfully, I added another commit for the cancellation, please see https://github.com/nextcloud/android/pull/547/commits/fa4439bc4619e00f984a878af11be055ff675479 and merge in case you are fine with it 👍

**Comment by tobiasKaminsky** [comment]
:+1:

**Pull request merged**